examples/modules.cpp

```diff
@@ -0,0 +1,121 @@
+#include <map>
+#include <string>
+
+#include <jsapi.h>
+
+#include <js/CompilationAndEvaluation.h>
+#include <js/Modules.h>
+#include <js/SourceText.h>
+
+#include "boilerplate.h"
+
+// This examples demonstrates how to compile ES modules in an embedding.
+//
+// See 'boilerplate.cpp' for the parts of this example that are reused in many
+// simple embedding examples.
+
+// Translates source code into a JSObject representing the compiled module. This
```

Comment:
This example is probably fine without it, but it might be good to explain what linking and instantiating means in this comment.